File: frontend/demo/component/combobox/combo-box-basic.ts
import '../../init'; // hidden-full-source-line
import '@vaadin/flow-frontend/comboBoxConnector'; // hidden-full-source-line

import { html, LitElement, customElement, property } from 'lit-element';
import '@vaadin/vaadin-combo-box/vaadin-combo-box';
import { Country } from '../../domain/Country';
import { getCountries } from '../../domain/DataService';

@customElement('combo-box-basic')
export class Example extends LitElement {
  @property({ type: Array })
  private items: Country[] = [];

  async firstUpdated() {
    this.items = await getCountries();
  }

  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-combo-box
        label="Country"
        item-label-path="name"
        item-value-path="id"
        .items="${this.items}"
      ></vaadin-combo-box>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/combobox/ComboBoxBasic.java
package com.vaadin.demo.component.combobox;

import com.vaadin.flow.component.combobox.ComboBox;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.Country;
import com.vaadin.demo.domain.DataService;

@Route("combo-box-basic")
public class ComboBoxBasic extends Div {

  public ComboBoxBasic() {
    // tag::snippet[]
    ComboBox<Country> comboBox = new ComboBox<>("Country");
    comboBox.setItems(DataService.getCountries());
    comboBox.setItemLabelGenerator(Country::getName);
    add(comboBox);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<ComboBoxBasic> { // hidden-source-line
  } // hidden-source-line
}
